public/app/plugins/panel/heatmap-new/utils.ts

```diff
@@ -919,7 +919,7 @@ export const boundedMinMax = (
 };
 
 export const valuesToFills = (values: number[], palette: string[], minValue: number, maxValue: number) => {
-  let range = maxValue - minValue;
+  let range = Math.max(maxValue - minValue, 1);
```

Comment:
this is unrelated, but was causing the mystery crash when we had only a single unique color, which made the range 0 and an empty i`ndexedFills` array.